### ISSUE TITLE
ElasticSearch does not support anchor operators, such as ^ (beginning of line) or $ (end of line)

### ISSUE THREAD

**Issue by kongnir** [react-querybuilder/react-querybuilder#742]
### React Query Builder version

v7.1.0

### Platform

Chrome on Mac

### Description

The conversion from Query Builder to ElasticSearch using `formatQuery(_query, "elasticsearch")` for `endsWith` creates the following result:

```
{
    "bool": {
        "must": [
            {
                "regexp": {
                    "email": {
                        "value": "gmail.com$"
                    }
                }
            }
        ]
    }
}
```

When we use the above generated DSL Query, ElasticSearch will not return any result. That's because in their [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/regexp-syntax.html), they mentioned that

_Lucene’s regular expression engine does not support anchor operators, such as ^ (beginning of line) or $ (end of line). To match a term, the regular expression must match the entire string._

### Proposed solution

If we still want to support `endsWith` and `beginsWith`, we will need to modify `formatQuery` such that it'll use wildcard instead of anchor operators.

Using the same example above, we should produce this DSL query instead:

```
{
    "bool": {
        "must": [
            {
                "regexp": {
                    "email": {
                        "value": ".*gmail.com"
                    }
                }
            }
        ]
    }
}
```

### Reproduction

Use the Query Builder to create a query with "endsWith" (example `gmail.com`), use `formatQuery(_query, "elasticsearch")` to convert it to DSL query. Feed that to ElasticSearch and it'll not return any result.

Now, create a query with "contains" (example `.*gmail.com`), use `formatQuery` to convert to DSL query. It will return some results.

### Expected behavior

Should return DSL query that is supported by Lucene's Regular Expressions (used by ElasticSearch).

### Additional information

Reference: https://www.elastic.co/guide/en/elasticsearch/reference/current/regexp-syntax.html

**Comment by jakeboone02** [comment]
Can you try v7.6.0? I think @flieks fixed this in #733.

**Comment by kongnir** [comment]
Thanks @jakeboone02 , I've tested v7.6.0 and it works very well. Cheers
